pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -717,24 +711,20 @@ func (c *InstallerController) newNodeStateForInstallInProgress(ctx context.Conte
 				staticPodErrors = append(staticPodErrors, fmt.Sprintf("no detailed termination message, see `oc get -oyaml -n %q pods %q`", ns, name))
 			}
 
-			ret := currNodeState.DeepCopy()
-			ret.LastFailedRevisionErrors = staticPodErrors
-
 			if startupMonitor {
-				// this will be overridden again when the fallback pod is observed
-				ret.LastFailedRevisionErrors = append(ret.LastFailedRevisionErrors, "falling back to last-known-good revision")
-			} else {
-				ret.TargetRevision = 0 // stop installer retries, but keep running in startup-monitor case where we leave this to the startup-monitor to do
-
-				// increase failed count, but not in startup-monitor case where this is done when the fallback pod is observed below in staticPodStateFallback,
-				ret.LastFailedRevision = currNodeState.TargetRevision
-				ret.LastFailedReason = nodeStatusOperandFailedReason
-				if ret.LastFailedTime == nil || !ret.LastFailedTime.Time.Equal(ts) {
-					// avoid increasing the count continuously by comparing the fail time
-					failTime := metav1.NewTime(ts)
-					ret.LastFailedTime = &failTime
-					ret.LastFailedCount++
-				}
+				return currNodeState, false, fmt.Sprintf("operand pod failed: %v", strings.Join(staticPodErrors, "\n")), nil
```

Comment:
I think we miss `staticPodErrors`, previously `ret.LastFailedRevisionErrors = staticPodErrors`

Reply:
intentional. Setting the error without the count is odd, no?